### PR TITLE
Add StorageClient class

### PR DESCRIPTION
**Pull request by mbarnes**
The storage service is used heavily by all other services.  This is a convenience API to make storage service methods less tedious to call.

It uses model instances as inputs and outputs (much like the old MVP API), providing consistent error handling and other behaviors like model validation prior to saving.

This should partially address https://github.com/projectatomic/commissaire-service/issues/30 but will benefit `commissaire-http` as well.

### CODE REVIEW

**Comment by mbarnes** [comment]
Hang on... still got a bit more work to do.

Got so focused on `commissaire-http` I forgot to write unit tests for this.

**Comment by mbarnes** [comment]
@ashcrow: :arrow_up: Okay, this is ready for review.

**Comment by ashcrow** [comment]
Reviewing now.

File: src/commissaire/storage/client.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+            return model_instance
+        except (RemoteProcedureCallError, models.ValidationError) as error:
+            self.bus_mixin.logger.error(
+                '{}: Unable to list {}s: {}'.format(
+                    self.bus_mixin.__class__.__name__,
+                    model_class.__name__, error))
+            raise error
+
+    # -----------------------------------------------------------------------
+    # Model-specific methods (esp. listable types)
```
Comment (ashcrow) [inline src/commissaire/storage/client.py:152]: Why are these 2 specific methods needed?

**Review by ashcrow — APPROVED** [review]
One question. Otherwise looks good.

File: src/commissaire/storage/client.py (hunk 1 above)
Comment (mbarnes) [reply src/commissaire/storage/client.py:152]: Not strictly needed, but I added them for convenience based on frequency of use.  There's a good many places were a host or cluster model is fetched with only an address or name string.

**Comment by mbarnes** [comment]
[(3) [multi-host-mgr-2] Decrease service complexity](https://trello.com/c/Gzjkpge2/561-3-multi-host-mgr-2-decrease-service-complexity)

**Pull request merged**